hrjobcontract/js/test/controllers/modal/modal-contract_test.js

```diff
@@ -0,0 +1,165 @@
+define([
+  'job-contract/app'
+], function() {
+  'use strict';
+
+  describe('ModalContractCtrl', function() {
+    var ctrl, $rootScope, $controller, $scope, $q, $uibModal, $httpBackend, $uibModalInstanceMock,
+      $uibModalMock, entity, ContractDetailsService;
+
+    beforeEach(module('hrjc'));
+
+    beforeEach(inject(function(_$controller_, _$rootScope_, _$httpBackend_, _$q_,
+      _ContractDetailsService_) {
+      $controller = _$controller_;
+      $rootScope = _$rootScope_;
+      $httpBackend = _$httpBackend_;
+      ContractDetailsService = _ContractDetailsService_;
+      $q = _$q_;
+    }));
+
+    beforeEach(function() {
+      $httpBackend.whenGET(/action=validatedates&entity=HRJobDetails/).respond({ success: true });
+      $httpBackend.whenGET(/action=get&entity=HRJobContract/).respond({});
+      $httpBackend.whenPOST(/action=create&entity/).respond({});
+      $httpBackend.whenPOST(/action=replace&entity/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+    });
+
+    beforeEach(function() {
+      mockObjects()
+      makeController();
+    });
+
+    beforeEach(function() {
+      callSpies();
+    });
+
+    describe("After call to save() function", function() {
+
+      beforeEach(function() {
+        $scope.save();
+        $rootScope.$digest();
+      });
+
+      it("calls to validate the dates form ContractDetailsService", function() {
+        expect(ContractDetailsService.validateDates).toHaveBeenCalled();
+
+        ContractDetailsService.validateDates().then(function(result) {
+          expect(result.success).toBe(true);
+        });
+      });
+
+      it("gets confirmation fron user to save contract data", function() {
+        expect($uibModalMock.open).toHaveBeenCalled();
+
+        $uibModalMock.open().result.then(function(data) {
+          expect(data).toBe('edit');
+        });
+      });
+
+      it("sets contract period_end_date to ''  if it was null", function() {
+        expect($scope.entity.details.period_end_date).toBe('');
+      });
+    });
+
+    describe("When period_end_date is undefined", function() {
```

Comment:
This describe block should be nested in the last describe block, as this test is part of `save()`